src/components/ContentOpenWith/IntegrationPortal.js

```diff
@@ -4,69 +4,51 @@
 import { createPortal } from 'react-dom';
 
 type Props = {
     children: any,
     integrationWindow: any,
 };
-class IntegrationPortal extends PureComponent<Props> {
-    containerElement: ?HTMLElement;
 
-    /**
-     * [constructor]
-     *
-     * @private
-     * @return {IntegrationPortal}
-     */
-    constructor(props: Props) {
-        super(props);
-
-        const { integrationWindow }: Props = this.props;
-        this.containerElement = integrationWindow.document.createElement('div');
-    }
-
-    /**
-     * Setup UI in the new window.
-     *
-     * @private
-     * @inheritdoc
-     * @return {void}
-     */
-    componentDidMount() {
-        const { integrationWindow }: Props = this.props;
-        this.copyStyles();
-        integrationWindow.document.body.appendChild(this.containerElement);
-    }
+/**
+ * Copies stylesheets to the new window.
+ *
+ * @private
+ * @return {void}
+ */
+const copyStyles = integrationWindow => {
+    // The new window will have no CSS, so we copy all style sheets as a safe way
+    // of ensuring required styles are present
+    Array.from(document.styleSheets).forEach(styleSheet => {
+        if (!styleSheet.href) {
+            return;
+        }
+
+        const copiedStyleSheet = integrationWindow.document.createElement(
+            'link',
+        );
+        copiedStyleSheet.rel = 'stylesheet';
+        copiedStyleSheet.href = styleSheet.href;
+        integrationWindow.document.head.appendChild(copiedStyleSheet);
+    });
+
+    // Reset margin and padding in our new window
+    integrationWindow.document.body.style.margin = 0;
+    integrationWindow.document.body.style.padding = 0;
+};
 
-    /**
-     * Copies stylesheets to the new window.
-     *
-     * @private
-     * @return {void}
-     */
-    copyStyles() {
-        const { integrationWindow }: Props = this.props;
-        // The new window will have no CSS, so we copy all style sheets as a safe way
-        // of ensuring required styles are present
-        Array.from(document.styleSheets).forEach(styleSheet => {
-            if (!styleSheet.href) {
-                return;
-            }
+const IntegrationPortal = ({ integrationWindow, children }: Props) => {
+    // If the integration window isn't closed, we should reuse the container div and avoid
+    // creating another one.
+    const existingContainer = integrationWindow.document.querySelector('div');
+    const containerElement =
+        existingContainer || integrationWindow.document.createElement('div');
 
-            const copiedStyleSheet = integrationWindow.document.createElement(
-                'link',
-            );
-            copiedStyleSheet.rel = 'stylesheet';
-            copiedStyleSheet.href = styleSheet.href;
-            integrationWindow.document.head.appendChild(copiedStyleSheet);
-        });
-    }
+    copyStyles(integrationWindow);
+    integrationWindow.document.body.appendChild(containerElement);
```

Comment:
does this account for if the integration window isn't closed, we don't append the containerElement again?

Reply:
good catch. Right now it will just re-append itself. Not sure if that actually changes anything but we probably shouldn't be doing it.